api/src/routes/settings.ts

```diff
@@ -651,3 +657,114 @@ ${isLinkSentWithinLimitTTL}`
 
   done();
 };
+
+/**
+ * Plugin for endpoints that redirect if the user is not authenticated.
+ *
+ * @param fastify The Fastify instance.
+ * @param _options Options for the plugin.
+ * @param done Callback to signal that the logic has completed.
+ */
+export const settingRedirectRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.addHook('onRequest', fastify.authorizeOrRedirect);
+
+  const redirectMessage = {
+    type: 'danger',
+    content:
+      'Oops! Something went wrong. Please try again in a moment or contact support@freecodecamp.org if the error persists.'
```

Comment:
We should think of storing these string somewhere as some of them are repeatedly used in many places